src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -192,12 +194,13 @@ internal override bool IsExpressionBodied
             MethodKind methodKind,
             bool isAutoPropertyAccessor,
             DiagnosticBag diagnostics) :
-            base(containingType, syntax.GetReference(), syntax.Body?.GetReference(), location)
+            base(containingType, syntax.GetReference(), syntax.Body?.GetReference() ?? syntax.ExpressionBody?.GetReference(), location)
         {
             _property = property;
             _explicitInterfaceImplementations = explicitInterfaceImplementations;
             _name = name;
             _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _isExpressionBodied = (syntax.Body == null && syntax.ExpressionBody != null) ? true : _property.IsExpressionBodied;
```

Comment:
I don't think I agree. If the accessor has a block as a body, it should not be considered as having an expression body (even if property has it). Other parts of the codebase should be adjusted if necessary.